### PR TITLE
Configure AMS quota backend

### PR DESCRIPTION
**Pull request by stehessel**
Configure AMS resource mappings for ACS. The AMS quota service is tested
by extensive unit tests using a mocked OCM service. We decided not to
switch the e2e-tests to AMS, and instead perform end-to-end tests
against AMS as part of a external probe (see
https://issues.redhat.com/browse/ROX-12076).

## Checklist (Definition of Done)

- [x] Unit and ~integration~ tests added
- [x] Documentation added if necessary
- [x] CI and all relevant tests are passing

## Test manual

- Deployed fleet manager with AMS quota type.
- Request Central creation.
- Verify OCM subscription is registered with correct billing model and quota is consumed.

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Review by dhaus67 — COMMENTED** [review]
Are we planning on enabling AMS quota types for the staging configuration under `internal/dinosaur/pkg/environment/staging.go`?
Also, by default we are using the ingeration instance of AMS (api.stage.openshift.com), is this also correct? I remember us having a conversation that this is only done for prod, but I could be mistaken.

I would also expect that we are setting the production values (AMS as quota type + prod AMS URL) in the production environment (under `internal/dinosaur/pkg/environment/production.go`).

**Comment by stehessel** [comment]
> Are we planning on enabling AMS quota types for the staging configuration under internal/dinosaur/pkg/environment/staging.go?
Also, by default we are using the ingeration instance of AMS (api.stage.openshift.com), is this also correct? I remember us having a conversation that this is only done for prod, but I could be mistaken.

I was planning to enable that, but tbh I wasn't aware of the environments module. I thought this was configured in app-interface rather than the fleet manager code itself. Let's have a quick chat about these environments (what is the difference between integration, testing, development and stage??). Currently I think defaulting to stage AMS is fine because it makes local testing easier. The prod fleet manager instance should use the AMS prod instance. The stage fleet manager is debatable - if we use AMS stage then we can be sure that the stage fleet manager does not consume quota of prod accounts. On the other hand, it's better have stage and prod as similar as possible. I'd probably lean toward both using AMS prod.

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 1]
```diff
@@ -29,8 +28,10 @@ import (
 	"github.com/stackrox/acs-fleet-manager/pkg/metrics"
 )
 
-var dinosaurDeletionStatuses = []string{constants2.DinosaurRequestStatusDeleting.String(), constants2.DinosaurRequestStatusDeprovision.String()}
-var dinosaurManagedCRStatuses = []string{constants2.DinosaurRequestStatusProvisioning.String(), constants2.DinosaurRequestStatusDeprovision.String(), constants2.DinosaurRequestStatusReady.String(), constants2.DinosaurRequestStatusFailed.String()}
+var (
+	dinosaurDeletionStatuses  = []string{constants2.DinosaurRequestStatusDeleting.String(), constants2.DinosaurRequestStatusDeprovision.String()}
+	dinosaurManagedCRStatuses = []string{constants2.DinosaurRequestStatusProvisioning.String(), constants2.DinosaurRequestStatusDeprovision.String(), constants2.DinosaurRequestStatusReady.String(), constants2.DinosaurRequestStatusFailed.String()}
```
[the remark this answers is not in the document]
Comment (stehessel) [reply internal/dinosaur/pkg/services/dinosaur.go]: Done

File: docs/quota/quota.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Quota control
+
+When fleet manager receives a creation request from a client, for example the Red Hat console,
+the OCM token attached to the request identifies the user. Fleet manager then determines if
```
[the remark this answers is not in the document]
Comment (stehessel) [reply docs/quota/quota.md]: Done

**Comment by stehessel** [comment]
The plan is:
- First test the end-to-end flow with the quota list backend to make sure we have a viable fallback.
- Then switch to AMS backend for staging and prod fleet manager.

So I'll still update the stage and prod environment to use the prod AMS instance, but we'll keep the quota list backend as default for this PR.

**Review by dhaus67 — APPROVED** [review]

**Pull request merged**